Add Avro Configuration Properties to Hive connector docs (#25311)
prestodb/presto: presto-docs/src/main/sphinx/connector/hive.rst

 ``hive.metastore.catalog.name``                   Specifies the catalog name to be passed to the metastore.
================================================== ============================================================ ============

Avro Configuration Properties
-----------------------------

> When querying or creating Avro-formatted tables via the Hive connector, you may need to supply or override the Avro schema. In addition, Hive Metastore (especially Hive 3.x) must be configured to read storage schemas for Avro tables. The following properties are relevant:

Reviewer: ```suggestion
When querying or creating Avro-formatted tables with the Hive connector, you may need to supply or override the Avro schema. In addition, Hive Metastore, especially Hive 3.x, must be configured to read storage schemas for Avro tables. 
```
Avoid via or or Latin terms and abbreviations in documentation. See the [GitLab style guide recommended word list entry for via](https://docs.gitlab.com/development/documentation/styleguide/word_list/#via) for discussion and alternates.

======================================================== ============================================================================== ======================================================================================
Property Name                                            Description                                                                    Default
======================================================== ============================================================================== ======================================================================================

> ``avro_schema_url``                                      URL or path (HDFS, S3, HTTP, etc.) to the Avro schema file for                 None (must be specified if Metastore does not provide or you need to

Reviewer: ```suggestion
``avro_schema_url``                                      URL or path (HDFS, S3, HTTP, or others) to the Avro schema file for                 None (must be specified if Metastore does not provide or you need to
```
[GitLab style guide recommended word list entry for etc.](https://docs.gitlab.com/development/documentation/styleguide/word_list/#etc)

                                                         reading an Avro-formatted table. If specified, Presto will fetch               override schema)
                                                         and use this schema instead of relying on any schema in the
                                                         Metastore.

> ``avro_schema_literal``                                  Inline JSON literal of the Avro schema to use when reading an                  None

Reviewer 2: Usually, we only document table properties that are supported while creating a table in Presto.  As far as I am aware, `avro_schema_literal` property is not supported in Presto.

                                                         Avro-formatted table. Useful when external schema files are not
                                                         accessible or for quick testing.

> ``metastore.storage.schema.reader.impl``                 Hive Metastore configuration property (to be set in hive-site.xml              DefaultStorageSchemaReader

Reviewer 2: I think we should have a separate section to highlight this, this isn't a table property like `avro_schema_url`

                                                         on the Metastore side). Specifies the class used to read storage               (``DefaultStorageSchemaReader.class.getName()``)

>                                                          schema (e.g., Avro schema). For Hive 3.x Metastore with Avro

Reviewer: ```suggestion
                                                         schema (such as Avro schema). For Hive 3.x Metastore with Avro
```
See [GitLab style guide recommended word list entry for e.g.](https://docs.gitlab.com/development/documentation/styleguide/word_list/#eg) for alternates.

                                                         tables, set to: ``org.apache.hadoop.hive.metastore.SerDeStorageSchemaReader``
                                                         to avoid "Storage schema reading not supported" errors.
======================================================== ============================================================================== ======================================================================================

> To support Avro tables with ``avro_schema_url`` when using Hive 3.x, you need to add the following property to ``hive-site.xml`` where the metastore service is running:

Reviewer: ```suggestion
To support Avro tables with ``avro_schema_url`` when using Hive 3.x, you must add the ``metastore.storage.schema.reader.impl`` property to ``hive-site.xml`` where the metastore service is running:
```

.. code-block:: xml

    <property>
      <name>metastore.storage.schema.reader.impl</name>
      <value>org.apache.hadoop.hive.metastore.SerDeStorageSchemaReader</value>
    </property>

> You will need to restart the metastore service for this configuration to take effect.

Reviewer: ```suggestion
You must restart the metastore service for this configuration to take effect.
```

Property Name                                            Description                                                                    Default
======================================================== ============================================================================== ======================================================================================
``avro_schema_url``                                      URL or path (HDFS, S3, HTTP, or others) to the Avro schema file for             None (must be specified if Metastore does not provide or you need to
                                                         reading an Avro-formatted table. If specified, Presto will fetch                override schema)
                                                         and use this schema instead of relying on any schema in the
                                                         Metastore.
======================================================== ============================================================================== ======================================================================================

Hive Metastore Configuration

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
The formatting line here must be exactly as many characters as the line above it that it is formatting.

To support Avro tables with schema properties when using Hive 3.x, you must configure the Hive Metastore service:

Add the ``metastore.storage.schema.reader.impl`` property to ``hive-site.xml`` where the metastore service is running:

.. code-block:: xml

    <property>
      <name>metastore.storage.schema.reader.impl</name>
      <value>org.apache.hadoop.hive.metastore.SerDeStorageSchemaReader</value>
    </property>

> You must restart the metastore service for this configuration to take effect. This setting allows the metastore to read storage schemas for Avro tables and avoids "Storage schema reading not supported" errors.

Reviewer: ```suggestion
You must restart the metastore service for this configuration to take effect. This setting allows the metastore to read storage schemas for Avro tables and avoids ``Storage schema reading not supported`` errors.
```
